rsocket/framing/ScheduledFrameProcessor.cpp

```diff
@@ -0,0 +1,26 @@
+#include "rsocket/framing/ScheduledFrameProcessor.h"
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/io/IOBuf.h>
+
+namespace rsocket {
+
+void ScheduledFrameProcessor::processFrame(
+    std::unique_ptr<folly::IOBuf> ioBuf) {
+  // ScheduledFrameProcessor could get destroyed before the scheduled
```

Comment:
You can get around this by submitting a copy of it only `~ScheduledFrameProcessor` gets called. Due to EventBases strict ordering you'll guarantee destruction doesn't happen until all callbacks have been processed.

This way we can avoid incrementing/decrementing the ref-count for every frame.

Reply:
💯